server/src/workspaceLookup.ts

```diff
@@ -0,0 +1,192 @@
+        });
+        // `onDidChangeWatchedFiles` changes are not fired by editing a file
+        // but when a file changes outside of the clients control (a git pull for example)
+        this.workspace.connection.onDidChangeWatchedFiles((params) => this.watchedFilesChanged(params));
+
+        return new Promise<void>((resolve, reject) => {
+
+            this.workspace.getWorkspaceFolders()
+                .then((folders) => {
+                    if (folders) {
+                        Promise.all(folders.map(folder => this.addFolder(folder)))
+                            .then(() => resolve())
+                            .catch(reason => reject(reason));
+                    } else {
+                        resolve();
+                    }
+                });
+        })
+    }
+
+    private collectionPromise<T>(array: T[], func: (param: T) => Promise<void>): Promise<void> {
+        const results: Promise<void>[] = [];
+        array.forEach(item => {
+            results.push(func(item));
+        });
+        return new Promise<void>((resolve, reject) => {
+            Promise.all(results)
+                .then(() => resolve())
+                .catch(reason => reject(reason));
+        });
+    }
+
+    private watchedFilesChanged(event: DidChangeWatchedFilesParams) {
+        // when a file watched by the client is changed even when it isn't open in the editor (by version control for example)
+        // this only happens with saved files and the changes should be read from the filesystem
+        return new Promise<void>((resolve, reject) => {
+            const parseResults: Promise<void>[] = [];
+
+            event.changes.forEach(change => {
+                if (change.uri.endsWith('.omt')) {
+                    switch (change.type) {
+                        case FileChangeType.Changed:
+                        case FileChangeType.Created:
+                            parseResults.push(parseOmtFile(change.uri.substr(7))
+                                .then(
+                                    (result) => this.workspaceModules.checkForChanges(result),
+                                    (reason) => { reject(reason) })
+                                .catch(reason => {
+                                    console.error(`while parsingOmtFile ${change.uri} something went wrong: ${reason}`);
+                                    reject(reason);
+                                }));
+                            break;
+                        case FileChangeType.Deleted:
+                            this.workspaceModules.removeFile(change.uri.substr(7));
+                            break;
+                        default:
+                            console.warn(`unsupported FileChangeType '${change.type}'`);
+                            break;
+                    }
+                }
+            });
+
+            Promise.all(parseResults).then(() => resolve());
```

Comment:
Is it crucial the promise returned be a `Promise<void>`?

Reply:
wow...
So, I traced the call back to a handler where the promise was no longer used either..
that means the whole promise construction of this function can be removed...
makes it a lot more simple to read